store/sqlstore/channel_store.go

```diff
@@ -1352,9 +1313,35 @@ func (s SqlChannelStore) SaveMultipleMembers(members []*model.ChannelMember) ([]
 	}
 	defer finalizeTransaction(transaction)
 
-	newMembers, appErr := s.saveMultipleMembersT(transaction, members)
-	if appErr != nil {
-		return nil, appErr
+	newMembers, err := s.saveMultipleMembersT(transaction, members)
+	if err != nil { // TODO: this will go away once SaveMultipleMembers is migrated too.
+		var cErr *store.ErrConflict
+		var intErr *store.ErrInternal
+		var appErr *model.AppError
+		switch {
+		case errors.As(err, &cErr):
+			switch cErr.Resource {
+			case "ChannelMembers":
+				return nil, model.NewAppError("CreateChannel", "store.sql_channel.save_member.exists.app_error", nil, cErr.Error(), http.StatusBadRequest)
+			}
+		case errors.As(err, &appErr): // in case we haven't converted to plain error.
+			return nil, appErr
+		case errors.As(err, &intErr):
+			switch {
+			case intErr.Location == "channel_roles_tosql":
```

Comment:
nit: can we change this `switch` to be a `switch intErr.Location` to keep consistency with the previous one?

Reply:
sounds good. Will do it after I merge the first one and fix the conflicts.